webui/react/src/pages/ExperimentList.tsx

```diff
@@ -80,13 +83,136 @@ const ExperimentList: React.FC = () => {
     },
   );
   const initSorter = storage.getWithDefault(STORAGE_SORTER_KEY, { ...defaultSorter });
-  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
-  const [ total, setTotal ] = useState(0);
+  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ filters, setFilters ] = useState<ExperimentFilters>(initFilters);
-  const [ sorter, setSorter ] = useState(initSorter);
+  const [ isUrlParsed, setIsUrlParsed ] = useState<boolean>(false);
+  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
   const [ search, setSearch ] = useState('');
-  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ selectedRowKeys, setSelectedRowKeys ] = useState<string[]>([]);
+  const [ sorter, setSorter ] = useState(initSorter);
+  const [ total, setTotal ] = useState(0);
+
+  /*
+   * When filters changes update the page URL.
+   */
+  useEffect(() => {
+    if (!isUrlParsed) return;
```

Comment:
If we remove this line, this should ensure the experiment list page will always have query params set right?

Reply:
That `isUrlParsed` is preventing a race condition between updating and parsing the query params.

1st load of the page we should try to parse it (and block updating), after the 1st render we should instead always update it (and avoid parsing it).